### DISCUSSION TITLE
Relation Queries works fine but gives typescript error of property does not exist on model.

### DISCUSSION THREAD

**Discussion by i701** [prisma/prisma#17462] [Q&A]
Hello, 
First of all before i get to the question i have read both [`relations`](https://www.prisma.io/docs/concepts/components/prisma-schema/relations) and [relation-queries](https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries) from the official documentation. From the documentation, i could not find anything related to solving this kind of an issue.

Here is the error i am getting.
 
![image](https://user-images.githubusercontent.com/72292943/213879149-2b16d34e-1ff2-4888-bc52-626ce4e0de11.png)

**Here is my `schema.prisma` file.**
```prisma
model Patient {
  id            Int          @id @default(autoincrement())
  firstName     String
  idCard        String       @unique
  sex           String       @default("MALE")
  address       String
  island        Island       @relation(fields: [islandId], references: [id])
  dob           DateTime
  phoneNumber   String
  bloodGroup    String       @default("O_POSITIVE")
  isThalassemia Boolean      @default(true)
  inCrossmatch  Boolean      @default(false)
  createdAt     DateTime     @default(now())
  updatedAt     DateTime     @updatedAt
  islandId      Int
  Crossmatch    Crossmatch[]

  @@map("patients")
}

model Crossmatch {
  id                  Int      @id @default(autoincrement())
  xmatchPatient       Patient  @relation(fields: [patientId], references: [id])
  xmatchDonor         Donor    @relation(fields: [donorId], references: [id])
  xmatchDonorHb       Float
  xmatchPatientHb     Float
  xmatchDate          DateTime @default(now())
  xmatchCompatibility String   @default("COMPATIBLE")
  xmatchStatus        String
  xmatchRejectReason  String?
  xmatchDoneBy        String

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt

  patientId Int
  donorId   Int
  Donation  Donation[]

  @@map("crossmatches")
}
```

There is no issue fetching the data using  [include](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#include) and i am able to get the data for `xmatchPatient` as you can see below.

![image](https://user-images.githubusercontent.com/72292943/213879418-77bf2e49-5d1c-4224-b273-e4fc8d763666.png)

Here is the `getServerSideProps` function.

![image](https://user-images.githubusercontent.com/72292943/213879502-26716e37-b7e5-4e6a-9817-e4f2f5d0a7d9.png)

i ran `npx prisma migrate` after creating the Crossmatch table and `npx prisma generate` as well. This error still persists and i don't understand why it did not generate the type for nested foreign key relationship. 
Did i do anything wrong here? Any help or some tips will be appreciate?

**Comment by nurul3101** [comment]
Hello @i701 👋 

The error that you are getting is expected as the generated types for a model only contain scalar fields by default, relations fieldsare not included. Here's a [guide](https://www.prisma.io/docs/concepts/components/prisma-client/advanced-type-safety/operating-against-partial-structures-of-model-types#problem-using-variations-of-the-generated-model-type) on how you can create types that include relation fields. Can you have a look at it? Let me know in case you still run into this issue.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
